src/v/kafka/server/handlers/handler.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+#include "kafka/protocol/types.h"
+#include "kafka/server/request_context.h"
+#include "kafka/server/response.h"
+#include "kafka/types.h"
+#include "utils/concepts-enabled.h"
+
+#include <seastar/core/smp.hh>
+
+namespace kafka {
+
+template<
+  typename RequestApi,
+  api_version::type MinSupported,
+  api_version::type MaxSupported>
+struct handler {
+    using api = RequestApi;
+    static constexpr api_version min_supported = api_version(MinSupported);
+    static constexpr api_version max_supported = api_version(MaxSupported);
+    static ss::future<response_ptr>
+    handle(request_context&&, ss::smp_service_group);
+};
+
+// clang-format off
+CONCEPT(
```

Comment:
since we only build on clang-11 and probably gcc-11 -- do we need the macro still ?

Reply:
mmm, i believe libc++ does not yet have concepts. so its disabled on clang+libc++ (clang+libstdc++ works ok).